### ISSUE TITLE
Flyout cant be closed when lost focus

### ISSUE THREAD

**Issue by wobushiafa** [AvaloniaUI/Avalonia#13629]
## Describe the bug
Logic Tree : Popup > Button > Flyout
the flyout cant close when loses the focus

## To Reproduce
I tried to change the ShowMode property of Flyout,but it didn't work

    <StackPanel>
        <Popup IsOpen="{Binding #Switch.IsChecked, Mode=OneWay}" Placement="Right">
            <Button Width="60"
                    Height="60"
                    Background="Black"
                    Content="Open Flyout">
                <Button.Flyout>
                    <Flyout ShowMode="Standard">
                        <Border Width="200"
                                Height="200"
                                Background="Red" />
                    </Flyout>
                </Button.Flyout>
            </Button>
        </Popup>
        <ToggleButton Name="Switch" Content="Open Popup" />
    </StackPanel>

## Screenshots
![1](https://github.com/AvaloniaUI/Avalonia/assets/80564298/1739ad61-4e0b-445c-a25e-b78ff8826a93)

## Environment 

- OS: Windows 11
- Avalonia-Version:11.0.5

**Comment by timunie** [comment]
Why is the Binding OneWay and not TwoWay? So if the popup closes, the ToggleButton should be unchecked imo.

**Comment by wobushiafa** [comment]
> Why is the Binding OneWay and not TwoWay? So if the popup closes, the ToggleButton should be unchecked imo.
I don't think Popup has any problems,its opened or closed by togglebutton,but the Flyout needs to be closed when it lose focus

**Comment by timunie** [comment]
yeah but it probably cannot as the binding tells it to stay open (it maybe cannot write back the value due to OneWay-Binding). So please give TwoWay at least a try. Thx.

**Comment by timunie** [comment]
Will also double-check on our end for sure.

**Comment by timunie** [comment]
```diff
    <StackPanel>
        <Popup IsOpen="{Binding #Switch.IsChecked, Mode=TwoWay}" Placement="Right"
+              IsLightDismissEnabled="True">
            <Button Width="60"
                    Height="60"
                    Background="Black"
                    Content="Open Flyout">
                <Button.Flyout>
                    <Flyout ShowMode="Standard">
                        <Border Width="200"
                                Height="200"
                                Background="Red" />
                    </Flyout>
                </Button.Flyout>
            </Button>
        </Popup>
        <ToggleButton Name="Switch" Content="Open Popup" />
    </StackPanel>
```

this works for me.

**Comment by wobushiafa** [comment]
> ```diff
>     <StackPanel>
>         <Popup IsOpen="{Binding #Switch.IsChecked, Mode=TwoWay}" Placement="Right"
> +              IsLightDismissEnabled="True">
>             <Button Width="60"
>                     Height="60"
>                     Background="Black"
>                     Content="Open Flyout">
>                 <Button.Flyout>
>                     <Flyout ShowMode="Standard">
>                         <Border Width="200"
>                                 Height="200"
>                                 Background="Red" />
>                     </Flyout>
>                 </Button.Flyout>
>             </Button>
>         </Popup>
>         <ToggleButton Name="Switch" Content="Open Popup" />
>     </StackPanel>
> ```
> 
> this works for me.

Popup should stay open

**Comment by wobushiafa** [comment]
> > ```diff
> >     <StackPanel>
> >         <Popup IsOpen="{Binding #Switch.IsChecked, Mode=TwoWay}" Placement="Right"
> > +              IsLightDismissEnabled="True">
> >             <Button Width="60"
> >                     Height="60"
> >                     Background="Black"
> >                     Content="Open Flyout">
> >                 <Button.Flyout>
> >                     <Flyout ShowMode="Standard">
> >                         <Border Width="200"
> >                                 Height="200"
> >                                 Background="Red" />
> >                     </Flyout>
> >                 </Button.Flyout>
> >             </Button>
> >         </Popup>
> >         <ToggleButton Name="Switch" Content="Open Popup" />
> >     </StackPanel>
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > this works for me.
> 
> Popup should stay open

click the popup ,the flyout can close,but click the window,the flyout cant close
![111111](https://github.com/AvaloniaUI/Avalonia/assets/80564298/e62715c2-0e1b-4861-b109-c3efbc4e5f72)

if flyout not in a popup,  the flyout can closed when click other area
![2222](https://github.com/AvaloniaUI/Avalonia/assets/80564298/ecbe662e-ac03-4469-8d53-73b53d701aa0)

**Comment by timunie** [comment]
Ah now I see. Sorry for the confusion.
